catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -203,27 +217,25 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             if (deleteResponse != null) {
                 List<Metacard> deletedMetacards = deleteResponse.getDeletedMetacards();
 
-                for (Metacard delete : deletedMetacards) {
-                    fileIdMap.delete(ingestedFile.getAbsolutePath());
-                    LOGGER.debug("content item deleted with id = {}", delete.getId());
+                if (safeKey != null) {
+                    fileIdMap.delete(safeKey);
                 }
+                logIds(deletedMetacards, "deleted");
             }
         }
     }
 
-    public void processHeaders(Map<String, Object> headers, StorageRequest storageRequest,
-            File ingestedFile) {
-        Map<String, Serializable> attributeOverrideHeaders = new HashMap<>((Map) headers.get(
-                Constants.ATTRIBUTE_OVERRIDES_KEY));
-        if (!attributeOverrideHeaders.isEmpty()) {
-            storageRequest.getProperties()
-                    .put(Constants.ATTRIBUTE_OVERRIDES_KEY,
-                            (Serializable) attributeOverrideHeaders);
-        }
-        if (headers.containsKey(Constants.STORE_REFERENCE_KEY)) {
-            storageRequest.getProperties()
-                    .put(Constants.STORE_REFERENCE_KEY, ingestedFile.getAbsolutePath());
-        }
+    protected void logIds(List<Metacard> metacards, String action) {
+        LOGGER.debug("content item(s) {} with id = {}",
```

Comment:
I was trying to be consistent due to the extra logic for the update case, but now that I think about it that will only ever have a size of one, so I think I will move the debug check into the helper method.